verifier/verifier.go

```diff
@@ -177,6 +179,18 @@ func (v *verifier) processSignature(ctx context.Context, sigBlob []byte, envelop
 			return err
 		}
 
+		pluginVersion := metadata.Version
+		var semVerRegEx = regexp.MustCompile(`^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$`)
+
+		//checking if the plugin version is in valid semver format
+		if !semVerRegEx.MatchString(pluginVersion) {
```

Comment:
Can we repalce it with https://pkg.go.dev/golang.org/x/mod/semver#IsValid

Reply:
We cannot use `isValid` method from `x/mod/semver` because of following reasons:
1. `isValid` requires versions to be prefixed with v which makes them invalid semver
2. `isValid` also allows partial strings such as 1, 1.0, etc. which again are invalid semver